vortex-serde/src/layouts/read/footer.rs

```diff
@@ -117,3 +114,53 @@ impl Footer {
             })
     }
 }
+
+pub struct FooterReader {
+    layout_serde: LayoutDeserializer,
+}
+
+impl FooterReader {
+    pub fn new(layout_serde: LayoutDeserializer) -> Self {
+        Self { layout_serde }
+    }
+
+    pub async fn read_footer<R: VortexReadAt>(
+        &self,
+        read: &R,
+        file_size: u64,
+    ) -> VortexResult<Footer> {
+        if file_size < FILE_POSTSCRIPT_SIZE as u64 {
+            vortex_bail!(
+                "Malformed vortex file, size {} must be at least {}",
+                file_size,
+                FILE_POSTSCRIPT_SIZE,
+            )
+        }
+
+        let read_size = INITIAL_READ_SIZE.min(file_size as usize);
+        let mut buf = BytesMut::with_capacity(read_size);
+        unsafe { buf.set_len(read_size) }
+
+        let read_offset = file_size - read_size as u64;
+        buf = read.read_at_into(read_offset, buf).await?;
+
+        let magic_bytes_loc = read_size - MAGIC_BYTES.len();
+
+        let magic_number = &buf[magic_bytes_loc..];
+        if magic_number != MAGIC_BYTES {
+            vortex_bail!("Malformed file, invalid magic bytes, got {magic_number:?}")
+        }
+
+        let ps = root::<vortex_flatbuffers::footer::Postscript>(
+            &buf[magic_bytes_loc - (FILE_POSTSCRIPT_SIZE - MAGIC_BYTES.len())..magic_bytes_loc],
+        )?;
+
+        Ok(Footer {
+            schema_offset: ps.schema_offset(),
+            footer_offset: ps.footer_offset(),
+            leftovers: buf.freeze(),
```

Comment:
I think this would be easier to read if you renamed `leftovers` to `initial_read_bytes` or some such (and maybe also added comments at the top on the semantics of the schema/layout offsets)